src/core/model/src/main/java/it/geosolutions/geostore/core/model/UserGroup.java

```diff
@@ -196,19 +196,65 @@ public String toString() {
         return builder.toString();
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#hashCode()
+     */
     @Override
-    public boolean equals(Object obj) {
-        if (this == obj) return true;
-        if (obj == null || getClass() != obj.getClass()) return false;
-        UserGroup other = (UserGroup) obj;
-        if (id != null && other.id != null) return id.equals(other.id);
-        // fallback if id is null (unsaved entities)
-        return groupName != null && groupName.equals(other.groupName);
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = (prime * result) + ((groupName == null) ? 0 : groupName.hashCode());
+        result = (prime * result) + ((id == null) ? 0 : id.hashCode());
+        result = (prime * result) + ((security == null) ? 0 : security.hashCode());
+        result = (prime * result) + ((attributes == null) ? 0 : attributes.hashCode());
+
+        return result;
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
     @Override
-    public int hashCode() {
-        if (id != null) return id.hashCode();
-        return groupName != null ? groupName.hashCode() : 0;
+    public boolean equals(Object obj) {
```

Comment:
Please do not touch this `equal` method. Groups must be compared by name only.

Reply:
not sure but this reverts this PR https://github.com/geosolutions-it/geostore/pull/444 can you please double check and make sure we don't have a regression?

Reply:
Reverted changes to `UserGroup` model entity.